main.js

```diff
@@ -154,6 +154,99 @@ if(fs.existsSync(__dirname + path.sep + ".." + path.sep + "OpenBazaar-Server" +
 // be closed automatically when the JavaScript object is GCed.
 var mainWindow = null;
 
+if (process.platform === "win32") {
+  initWin32();
+}
+
+function initWin32() {
+  var Registry = require('winreg');
+
+  var iconPath = path.join(process.env.LOCALAPPDATA, "app.ico");
+  registerProtocolHandlerWin32('ob', 'URL:OpenBazaar URL', iconPath, process.execPath);
+
+  /**
+   * To add a protocol handler, the following keys must be added to the Windows registry:
+   *
+   * HKEY_CLASSES_ROOT
+   *   $PROTOCOL
+   *     (Default) = "$NAME"
+   *     URL Protocol = ""
+   *     DefaultIcon
+   *       (Default) = "$ICON"
+   *     shell
+   *       open
+   *         command
+   *           (Default) = "$COMMAND" "%1"
+   *
+   * Source: https://msdn.microsoft.com/en-us/library/aa767914.aspx
+   *
+   * However, the "HKEY_CLASSES_ROOT" key can only be written by the Administrator user.
+   * So, we instead write to "HKEY_CURRENT_USER\Software\Classes", which is inherited by
+   * "HKEY_CLASSES_ROOT" anyway, and can be written by unprivileged users.
+   */
+
+  function registerProtocolHandlerWin32 (protocol, name, icon, command) {
+    var protocolKey = new Registry({
+      hive: Registry.HKCU, // HKEY_CURRENT_USER
+      key: '\\Software\\Classes\\' + protocol
+    })    
+
+    setProtocol()
+
+    function setProtocol (err) {
+      if (err) log.error(err.message)
```

Comment:
If there's an error, should it continue with the rest of the function, or do a return?